services/datalad/datalad_service/common/asyncio.py

```diff
@@ -2,14 +2,23 @@
 import subprocess
 
 
-async def run_check(command, dataset_path):
+async def run_check(command, dataset_path, env=None):
     """Helper to run an async command and check for failure"""
-    process = await asyncio.create_subprocess_exec(
-        *command,
-        cwd=dataset_path,
-        stdout=asyncio.subprocess.PIPE,
-        stderr=asyncio.subprocess.PIPE,
-    )
+    if env:
+        process = await asyncio.create_subprocess_exec(
+            *command,
+            cwd=dataset_path,
+            env=env,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE,
+        )
+    else:
+        process = await asyncio.create_subprocess_exec(
+            *command,
+            cwd=dataset_path,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE,
+        )
```

Comment:
The default value of `env` is already `None`: https://docs.python.org/3/library/subprocess.html#subprocess.Popen

```suggestion
    process = await asyncio.create_subprocess_exec(
        *command,
        cwd=dataset_path,
        env=env,
        stdout=asyncio.subprocess.PIPE,
        stderr=asyncio.subprocess.PIPE,
    )
```

Did you find this made a difference in practice?